core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java

```diff
@@ -21,6 +21,7 @@
 	}
 
 	public Waiter hasBuildCompleted(String buildConfigName) {
+		isLatestBuildPresent(buildConfigName).waitFor();
```

Comment:
Actual waiting goes against convention of this class. None of the methods waits for anything, they only return executable Waiter objects.

Reply:
ok.. Need to set the documentation then.

Reply:
I will update this

Reply:
Updating the docs is good but you can also fix the method so the user doesn't have to call `isLatestBuildPresent` and then this one by rewriting the Waiter implementation for this method. It'll be combination of the one isLatestBuildPresent and the one that it calls right now.

Reply:
ok. Did a simple implementation using Optional and returning null string if not yet existing.

Could also creates a SimpleWaiter with the condition with condition not null and Phase Complete or Failed but thought it clearer and in sync with the other 'hasBuildCompleted' method. Let me know what you think of it.